packages/ruby/lib/readme/har_request.rb

```diff
@@ -0,0 +1,79 @@
+require "rack/request"
+
+module Readme
+  class HarRequest
+    HTTP_NON_HEADERS = [
+      Rack::HTTP_COOKIE,
+      Rack::HTTP_VERSION,
+      Rack::HTTP_HOST,
+      Rack::HTTP_PORT
+    ]
+
+    def initialize(env)
+      @env = Rack::Request.new(env)
+    end
+
+    def as_json
+      {
+        method: @env.request_method,
+        queryString: to_hash_array(@env.GET),
+        url: @env.url,
+        httpVersion: http_version,
+        headers: http_headers,
+        cookies: cookies,
+        postData: {
+          text: request_body,
+          mimeType: @env.content_type
+        },
+        headersSize: -1,
+        bodySize: @env.content_length.to_i
+      }
+    end
+
+    def cookies
+      to_hash_array(@env.cookies)
+    end
+
+    def http_version
+      @env.get_header("HTTP_VERSION")
```

Comment:
Exposing these public methods feels weird. Perhaps this object is playing two roles: Serializing HAR requests, and modeling an HTTP request. I wonder if maybe I should extract another object here. As a bonus, if that's injected into `HarRequest`, then I can probably pass in a nice double rather than having to construct an env hash every time 🤔

Reply:
I see what you mean, but it doesn't feel too off to me. Extracting another object does sound like it'll clean up testing nicely.